libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/tasks/Tasks.kt

```diff
@@ -288,15 +288,17 @@ abstract class AbstractKotlinCompile<T : CommonCompilerArguments>() : AbstractKo
     }
 
     @get:Internal // takes part in the compiler arguments
-    val friendPaths: Array<String> by project.provider {
-        taskData.compilation.run {
-            if (this !is AbstractKotlinCompilation<*>) return@run emptyArray<String>()
-            associateWithTransitiveClosure
-                .flatMap { it.output.classesDirs }
-                .plus(friendArtifacts)
-                .map { it.absolutePath }.toTypedArray()
+    val friendPaths: FileCollection = project.files(
+        project.provider {
```

Comment:
Is "project.provider" necessary here?

Reply:
I am not sure when exactly is `associateWithTransitiveClosure` populated, so let's keep it for now and we can do a cleanup of friend paths later.